app/admin/camps.rb

```diff
@@ -2,12 +2,12 @@
   EXCLUDED = %w(contact_email user_id seeking_members safetybag_firstMemberName safetybag_firstMemberEmail safetybag_secondMemberName safetybag_secondMemberEmail)
 
   scope :active, default: true do |dreams|
-      dreams.active(true)
+      dreams.displayed.where(active: true)
     end
   scope :displayed_with_tags, default: false
-  scope("Public") { |scope| scope.where(is_public: true) }
-  scope("Private") { |scope| scope.where(is_public: false) }
-  scope("Inactive") { |scope| scope.where(active: false) }
+  scope("Public") { |dreams| dreams.displayed.where(is_public: true) }
```

Comment:
Can you please explain? (not sure why `displayed` has to be added now)

Reply:
OK! Makes sense. 
The reason why you need WHERE (and ON won't do the trick) is because for this purpose, an INNER JOIN would be what you need (get all rows that contain data from both tables, see http://www.complexsql.com/sql-joins-2/).
Seeing this is the requirement, you can actually use ActiveRecord's `joins` with `camp_manager` statements to do this for you (`camps.joins(:camp_manager)`).

Reply:
@sh0oki I have updated to use `joins` as suggested, then merged.